recipes/eigen/build.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+mkdir -p build
+cd build
+
+cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} -DCMAKE_BUILD_TYPE=Release ..
+make check
```

Comment:
`-j$CPU_COUNT` to speed things up? Although I'm not sure that will help CircleCI since I assume we only get given one core - but it's a lot quicker locally!